service/scaladsl/api/src/main/scala/com/lightbend/lagom/scaladsl/api/ServiceInfo.scala

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.scaladsl.api
+
+sealed trait ServiceInfo {
+  /**
+   * The name of this service.
+   */
+  val serviceName: String
+}
+
+object ServiceInfo {
+
+  def apply(name: String): ServiceInfo = ServiceInfoImpl(name)
+
+  private case class ServiceInfoImpl(serviceName: String) extends ServiceInfo
+}
```

Comment:
Straight forward example of the public sealed treat private case class pattern I'm using.